### PR TITLE
[TACHYON-911] Migrated the Web servers to use the new masters

### PR DESCRIPTION
**Pull request by yupeng9**
https://tachyon.atlassian.net/browse/TACHYON-911

### CODE REVIEW

File: servers/src/main/java/tachyon/master/next/MasterInfo.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+  /**
+   * Gets the list of block info of an InodeFile determined by path.
+   *
+   * @param path path to the file
+   * @return The list of the block info of the file
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public List<BlockInfo> getBlockInfoList(TachyonURI path)
```
Comment (gpang) [inline servers/src/main/java/tachyon/master/next/MasterInfo.java]: Why does this method exist? If block info is required for a file, it should be reading using FileBlockInfo.

File: servers/src/main/java/tachyon/master/next/MasterInfo.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+  /**
+   * Gets the list of blocks of an {@link InodeFile} determined by the file id.
+   *
+   * @param fileId id of the file
+   * @return The list of the blocks of the file
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public List<FileBlockInfo> getFileBlockList(long fileId)
```
Comment (gpang) [inline servers/src/main/java/tachyon/master/next/MasterInfo.java]: The web ui needs getFileBlockList() of both an id and a path?

File: servers/src/main/java/tachyon/master/next/MasterInfo.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+  /**
+   * Gets the file info for the file at the given path
+   *
+   * @param path path of the file
+   * @return the file info
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public FileInfo getFileInfo(TachyonURI path)
```
Comment (gpang) [inline servers/src/main/java/tachyon/master/next/MasterInfo.java]: The web ui needs getFileInfo() of both an id and a path?

File: servers/src/main/java/tachyon/master/next/filesystem/FileSystemMaster.java [hunk 4]
```diff
@@ -272,6 +275,28 @@ public boolean isDirectory(long id) {
...
 
+  /**
+   * Gets the list of block info of an InodeFile determined by path.
+   *
+   * @param path path to the file
+   * @return The list of the block info of the file
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public List<BlockInfo> getBlockInfoList(TachyonURI path)
```
Comment (gpang) [inline servers/src/main/java/tachyon/master/next/filesystem/FileSystemMaster.java]: Shouldn't the fs matser return FileBlockInfo? And, why can't this re-use getFileBlockInfoList()?

**Comment by gpang** [comment]
Thanks for the PR! I'm assuming the servlets are more or less ported from the old version?

File: servers/src/main/java/tachyon/master/next/MasterInfo.java (hunk 1 above)
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/next/MasterInfo.java]: Good question. I ported this from the old MasterInfo, which returns blockInfo to WebInterfaceBrowseServlet to build UiBlockInfo for display. My take is that if WebInterfaceBrowseServlet can display all kinds of blocks, e.g. file block, table block, etc, then we'd better use blockInfo for uniformity.

File: servers/src/main/java/tachyon/master/next/MasterInfo.java (hunk 2 above)
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/next/MasterInfo.java]: Good call. I added them for consistency. The web ui uses fileId, but I found some other methods use path, e.g. getBlockInfoList, so I added both. What do you think? I can retain only the necessary ones, and add back others when needed.

File: servers/src/main/java/tachyon/master/next/MasterInfo.java (hunk 3 above)
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/next/MasterInfo.java]: yes

File: servers/src/main/java/tachyon/master/next/filesystem/FileSystemMaster.java (hunk 4 above)
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/next/filesystem/FileSystemMaster.java]: Same reason: it's used by WebInterfaceBrowseServlet for constructing uiBlockInfo

**Comment by yupeng9** [comment]
Yes, most servlets are simply ported from the old version, and I made some minor changes in addition to the necessary parts for using the new master info. It's unfortunate that git cannot track the diffs on file copies, like *svn copy*.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by gpang** [comment]
LGTM

**Pull request merged**